### DISCUSSION TITLE
Unable to get it functioning in any capacity with Unreal engine

### DISCUSSION THREAD

**Discussion by patton3** [JSBSim-Team/jsbsim#965] [Q&A]
Hi, I'd made another post but since then I've had even more issues and tried more solutions. Following the UnrealEngine instructions to the letter I'm unable to get JSBSimForUnreal.sln to build without failing despite using VS2019 and 2022 with the correct plugins. After a dozen attempts it worked randomly and I've been using those files ever since to try and get the UEReferenceApp.uproject working. Building it automatically gives an error and tells me to build it manually, following the README steps to do that gives further errors and fails. Removing the required Windows toolchain version (14.29.30136) will sometimes generate the UEReferenceApp.sln file although after working once it has failed every time since when trying to regenerate. Trying to "Build Startup project" with the UnrealVS plugin will have 9 successes and 1 failure, but the UEReferenceApp.uproject still does not function. I've tried it on several computers, using every Unreal version since 5.0 and every Visual studio toolset even plausibly likely to work on top of the ones required in the documentation. I can provide errors if requested but there are so many I was wondering if anyone has tried it recently, maybe the Git build is broken, or if not I'd love some help.

**Comment by patton3** [comment]
Ok, while I don't have a solution for why JSBSimForUnreal.sln doesn't work, using the files from the one time it did I discovered that there is an extraneous " ||" at line 12 column 81 in jsbsim-master\UnrealEngine\Plugins\JSBSimFlightDynamicsModel\Source\ThirdParty\JSBSim.Build.cs
Once removed you are able to generate Visual Studio project files successfully and it runs UEReferenceApp.sln with the "Build Startup Project" option. This allows the whole example project to run.

**Reply by agodemar** [reply]
@patton3 if you are working on your own fork, and you confirm that this "||" causes the problem, you can submit pull request to the main repository.

**Reply by bcoconni** [reply]
Hmm, I'm a bit surprised by this error report. I'm by no means a C# programmer but the code at stake looks very much like C++. And what bothers me is that this is a multiline boolean expression so if the `||` token is removed on line 12, I guess the C# compiler will report an error ?
https://github.com/JSBSim-Team/jsbsim/blob/c9b12c4794046ae1406e1a66941937a64417bfd5/UnrealEngine/Plugins/JSBSimFlightDynamicsModel/Source/ThirdParty/JSBSim.Build.cs#L12-L15
@demonixis what is your opinion ?

**Reply by demonixis** [reply]
Can you elaborate on what you've done? This operator is required and there is no sens to use only one pipe. Please submit a log message.
